TimeOfFlight/src/TOFUtils.cc

```diff
@@ -28,42 +29,54 @@ using dd4hep::rec::Vector3D;
 using dd4hep::rec::FixedPadSizeTPCData;
 using CLHEP::RandGauss;
 
-
-dd4hep::rec::Vector3D TOFUtils::getHelixMomAtTrackState(const EVENT::TrackState& ts, double bField){
-    double phi = ts.getPhi();
-    double d0 = ts.getD0();
-    double z0 = ts.getZ0();
-    double omega = ts.getOmega();
-    double tanL = ts.getTanLambda();
-
-    HelixClass helix;
-    helix.Initialize_Canonical(phi, d0, z0, omega, tanL, bField);
-    return helix.getMomentum();
-}
-
-
-double TOFUtils::getTPCOuterR(){
-    auto& detector = Detector::getInstance();
-    DetElement tpcDet = detector.detector("TPC");
-    FixedPadSizeTPCData* tpc = tpcDet.extension <FixedPadSizeTPCData>();
-    return tpc->rMaxReadout/dd4hep::mm;
+EVENT::TrackerHit* TOFUtils::getSETHit(EVENT::Track* track){
+    vector<TrackerHit*> hits = track->getTrackerHits();
+    auto isSETHit = [](TrackerHit* hit) -> bool {
+        UTIL::BitField64 encoder( UTIL::LCTrackerCellID::encoding_string() ) ;
```

Comment:
I am not entirely sure (would need to be profiled / benchmarked), but I am not sure if the compiler sees this as something that he can inline (and in the end hoist outside the loop), so that he doesn't construct an encoder in each call in `find_if` below. Without benchmarking, the easiest thing to do probably would be to create the encoder outside the lambda and simply capture it by reference.